pkg/app/config.go

```diff
@@ -74,6 +82,34 @@ func CreateConfig(configFile string) (Config, error) {
 		config.Openshift.Namespace = string(dat)
 	}
 
+	for regCount, reg := range config.Registry {
+		if reg.AuthType == "secret" {
+			data, err = clients.GetSecretData(reg.AuthName, config.Openshift.Namespace)
+			if err != nil {
+				return Config{}, fmt.Errorf("Failed to find Dockerhub credentials in secret: %s", reg.AuthName)
+			}
+			var username = strings.TrimSpace(string(data["username"]))
+			var password = strings.TrimSpace(string(data["password"]))
+
+			if username == "" || password == "" {
+				return Config{}, fmt.Errorf("Secret: %s did not contain username/password credentials", reg.AuthName)
+			}
+
+			config.Registry[regCount].User = username
+			config.Registry[regCount].Pass = password
+
+		} else if reg.AuthType == "file" {
+			if dat, err = ioutil.ReadFile(reg.AuthName); err != nil {
+				return Config{}, err
+			}
+			if err = yaml.Unmarshal(dat, &regCred); err != nil {
```

Comment:
The logging above here is really good. We could improve the logging here by adding a message about what failed. Could be something like 'missing or invalid username/password'.